File: src/main/java/com/fifo/ticketing/domain/user/mapper/UserMapper.java
package com.fifo.ticketing.domain.user.mapper;

import com.fifo.ticketing.domain.user.dto.UserDto;
import com.fifo.ticketing.domain.user.entity.User;
import org.springframework.data.domain.Page;

public class UserMapper {

    private UserMapper() {
    }

    public static UserDto toUserDto(User user) {
        return new UserDto (
            user.getId(),
            user.getUsername(),
            user.getEmail(),
            user.isBlocked(),
            user.getRole()
            );
    }

    public static Page<UserDto> toUserDtoPage(Page<User> users) {
        return users.map(UserMapper::toUserDto);
    }

}

File: src/main/java/com/fifo/ticketing/domain/user/mapper/UserMapper.kt
package com.fifo.ticketing.domain.user.mapper

import com.fifo.ticketing.domain.user.dto.UserDto
import com.fifo.ticketing.domain.user.entity.User
import org.springframework.data.domain.Page

object UserMapper {

    @JvmStatic
    fun toUserDto(user: User): UserDto {
        return UserDto(
            user.id,
            user.username,
            user.email,
            user.isBlocked,
            user.role
        )
    }

    @JvmStatic
    fun toUserDtoPage(users: Page<User>): Page<UserDto> {
        return users.map { toUserDto(it) }
    }
}
